plugins/Marketplace/vue/src/StartFreeTrial/StartFreeTrial.vue

```diff
@@ -5,103 +5,318 @@
+      return translate(
+        'Marketplace_TrialStartNoLicenseLegalHint',
+        externalLink('https://shop.matomo.org/terms-conditions/'),
+        '</a>',
+        externalLink('https://matomo.org/privacy-policy/'),
+        '</a>',
+      );
+    },
+  },
+  methods: {
+    closeModal() {
+      $('#startFreeTrial').modal('close');
+    },
+    createAccountAndStartFreeTrial() {
+      if (!this.createAccountEmail) {
+        return;
+      }
 
-      window.Piwik_Popover.showLoading('');
+      this.showLoadingModal(true);
 
       AjaxHelper.post(
         {
           module: 'API',
-          method: 'Marketplace.startFreeTrial',
+          method: 'Marketplace.createAccount',
+        },
+        {
+          email: this.createAccountEmail,
+        },
+        {
+          createErrorNotification: false,
         },
-        { pluginName },
       ).then(() => {
-        window.Piwik_Popover.close();
-
-        const notificationInstanceId = NotificationsStore.show({
-          message: translate(
-            'CorePluginsAdmin_PluginFreeTrialStarted',
-            '<strong>',
-            '</strong>',
-            pluginName,
-          ),
-          context: 'success',
-          id: 'startTrialSuccess',
-          type: 'transient',
-        });
-        NotificationsStore.scrollToNotification(notificationInstanceId);
-        this.$emit('trialStarted');
+        this.startFreeTrial();
       }).catch((error) => {
-        window.Piwik_Popover.showError('', error.message);
-      }).finally(() => this.$emit('update:modelValue', ''));
+        if (error.message.startsWith('Marketplace_CreateAccountError')) {
+          this.showErrorModal(translate(error.message));
+
+          this.trialStartInProgress = false;
+
+          this.$emit('update:modelValue', '');
+        } else {
+          this.createAccountError = error.message;
+
+          this.trialStartInProgress = false;
+
+          this.showLicenseDialog(true);
+        }
+      });
     },
-    showLicenseDialog() {
-      $('#trialRequiresLicense').modal({
+    showLicenseDialog(immediateTransition: boolean) {
+      const onEnter = (event: KeyPressEvent) => {
+        const keycode = event.keyCode ? event.keyCode : event.which;
+        if (keycode === 13) {
+          this.closeModal();
+          this.createAccountAndStartFreeTrial();
+        }
+      };
+
+      const modalOptions: ModalOptions = {
         dismissible: true,
+        onOpenEnd: () => {
+          const emailField = '.modal.open #email';
+          $(emailField).focus();
+          $(emailField).off('keypress').keypress(onEnter);
+        },
         onCloseEnd: () => {
+          this.createAccountError = null;
+
+          if (this.trialStartInProgress) {
+            return;
+          }
+
           this.$emit('update:modelValue', '');
         },
+      } as unknown as ModalOptions;
+
+      if (immediateTransition) {
+        modalOptions.inDuration = 0;
```

Comment:
The previous construction with the ternary operator was breaking the onOpenEnd callback causing it to never be called for the undefined value, so it's best if the config option is not set at all when not needed.